tests/testthat/test_ncdf.R

```diff
@@ -40,12 +51,15 @@ test_that("normal bcsd", {
   expect_equal(st_dim$longitude$delta, 0.125)
 })
 
-# Leaving these here but commented in case there is interest in more robust handling of odd files.
-# see dblodgett-usgs/stars for sample files or PR #87 in r-spatial/stars.
-# test_that("broken bcsd", {
-#   expect_warning(nc <- read_ncdf(system.file("nc/bcsd_obs_1999_borked.nc", package = "stars")),
-#                  "Found non-canonical axis order in NetCDF unexpected bahavior may result.")
-# })
+test_that("non canonical axis order is handled right", {
+  expect_warning(nc <- read_ncdf(system.file("nc/3B42_Daily.19991231.7.test.nc", 
+                                             package = "stars")),
+                 "Non-canonical axis order found, attempting to correct.")
```

Comment:
To suppress the missing variable message, I tried
```
nc <- read_ncdf(system.file("nc/3B42_Daily.19991231.7.test.nc", var = "precipitation",
  package = "stars"))
# Error in nc_meta.character(.x) : NetCDF source cannot be empty string
```
is that an error?

Reply:
try: `nc <- read_ncdf(system.file("nc/3B42_Daily.19991231.7.test.nc"), var = "precipitation",
  package = "stars")` -- `()` error.

Reply:
thanks, my bad.